scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/listener/JobTaskStatusListener.java

```diff
@@ -0,0 +1,181 @@
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import org.objectweb.proactive.annotation.ImmediateService;
+import org.objectweb.proactive.api.PAActiveObject;
+import org.ow2.proactive.scheduler.common.NotificationData;
+import org.ow2.proactive.scheduler.common.Scheduler;
+import org.ow2.proactive.scheduler.common.SchedulerEvent;
+import org.ow2.proactive.scheduler.common.SchedulerEventListener;
+import org.ow2.proactive.scheduler.common.job.*;
+import org.ow2.proactive.scheduler.common.task.TaskInfo;
+import org.ow2.proactive.scheduler.common.task.TaskStatus;
+
+
+public class JobTaskStatusListener implements SchedulerEventListener, Serializable {
+    public static final int WAIT_TIME = 1000;
+
+    private List<TimestampedData<JobInfo>> jobEvents = new CopyOnWriteArrayList<>();
+
+    private List<TimestampedData<TaskInfo>> taskEvents = new CopyOnWriteArrayList<>();
+
+    public static JobTaskStatusListener attachListener(Scheduler scheduler) throws Exception {
+        JobTaskStatusListener listener = new JobTaskStatusListener();
+        listener = PAActiveObject.turnActive(listener);
+        scheduler.addEventListener(listener, false);
+        return listener;
+    }
+
+    @Override
+    public void jobStateUpdatedEvent(NotificationData<JobInfo> notificationData) {
+        final JobInfo jobInfo = notificationData.getData();
+        final TimestampedData<JobInfo> timestampedData = new TimestampedData<>(jobInfo);
+        jobEvents.add(timestampedData);
+    }
+
+    @Override
+    public void taskStateUpdatedEvent(NotificationData<TaskInfo> notificationData) {
+        final TaskInfo taskInfo = notificationData.getData();
+        final TimestampedData<TaskInfo> timestampedData = new TimestampedData<>(taskInfo);
+        taskEvents.add(timestampedData);
+    }
+
+    @ImmediateService
+    public long blockToGetJobTimestamp(JobId jobId, JobStatus jobStatus) throws Exception {
+        return blockToGetJobTimestampInternal(jobId, jobStatus, 0);
+    }
+
+    @ImmediateService
+    public long blockToGetJobTimestamp(JobId jobId, JobStatus jobStatus, long filter) throws Exception {
+        return blockToGetJobTimestampInternal(jobId, jobStatus, filter);
+    }
+
+    @ImmediateService
+    public long blockToGetTaskTimestamp(JobId jobId, TaskStatus taskStatus) throws Exception {
+        return blockToGetTaskTimestampInternal(jobId, taskStatus, 0);
+
+    }
+
+    @ImmediateService
+    public long blockToGetTaskTimestamp(JobId jobId, TaskStatus taskStatus, long filter) throws Exception {
+        return blockToGetTaskTimestampInternal(jobId, taskStatus, filter);
+    }
+
+    @Override
+    public void schedulerStateUpdatedEvent(SchedulerEvent eventType) {
+
+    }
+
+    @Override
+    public void jobSubmittedEvent(JobState job) {
+
+    }
+
+    @Override
+    public void jobUpdatedFullDataEvent(JobState job) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)